server/neptune/workflows/internal/deploy/terraform/state.go

```diff
@@ -12,86 +12,79 @@ import (
 	"go.temporal.io/sdk/workflow"
 )
 
-type activity interface {
+type receiverActivities interface {
 	UpdateCheckRun(ctx context.Context, request activities.UpdateCheckRunRequest) (activities.UpdateCheckRunResponse, error)
 }
 
 type StateReceiver struct {
-	ctx      workflow.Context
-	repo     github.Repo
-	activity activity
+	Repo     github.Repo
+	Activity receiverActivities
 }
 
-func NewStateReceiver(ctx workflow.Context, repo github.Repo, a activity) *StateReceiver {
-	return &StateReceiver{
-		ctx:      ctx,
-		repo:     repo,
-		activity: a,
-	}
-}
-
-func (n *StateReceiver) Receive(c workflow.ReceiveChannel, checkRunID int64) {
+func (n *StateReceiver) Receive(ctx workflow.Context, c workflow.ReceiveChannel, checkRunID int64) {
 	var workflowState *state.Workflow
-	c.Receive(n.ctx, &workflowState)
+	c.Receive(ctx, &workflowState)
 
 	// TODO: if we never created a check run, there was likely some issue, we should attempt to create it again.
 	if checkRunID == 0 {
-		logger.Error(n.ctx, "check run id is 0, skipping update of check run")
+		logger.Error(ctx, "check run id is 0, skipping update of check run")
 		return
 	}
 
 	// this shouldn't be possible
 	if workflowState.Plan == nil {
-		logger.Error(n.ctx, "Plan job state is nil, This is likely a bug. Unable to update checks")
+		logger.Error(ctx, "Plan job state is nil, This is likely a bug. Unable to update checks")
 		return
 	}
 
 	summary := markdown.RenderWorkflowStateTmpl(workflowState)
 	checkRunState, checkRunConclusion := determineCheckRunStateAndConclusion(workflowState)
 
 	// cap our retries for non-terminal states to allow for at least some progress
-	ctx := n.ctx
 	if checkRunState != github.CheckRunComplete {
-		ctx = workflow.WithRetryPolicy(n.ctx, temporal.RetryPolicy{
+		ctx = workflow.WithRetryPolicy(ctx, temporal.RetryPolicy{
 			MaximumAttempts: 3,
 		})
 	}
 
 	// TODO: should we block here? maybe we can just make this async
```

Comment:
can we lose ordering of activity runs if we don't block?

Reply:
Possibly, but I'm not sure, that's why im leaving as sync for now.